File: main workshop/services/dotnet/payment-processor/PaymentProcessor.cs
/*
 * This file is part of the Payment Processor service.
 * Update this file in the Module 2 Idempotency exercise.
 */

public class PaymentProcessor
{
    private readonly PaymentService _paymentService;

    public PaymentProcessor()
    {
        _paymentService = new PaymentService();
    }

    public async Task<PaymentResult> HandlePaymentAsync(DriverAssignedEvent driverEvent)
    {
        var result = await _paymentService.ProcessPaymentAsync(driverEvent);

        if (result.Success)
        {
            Logger.AppendKey("PaymentAmount", result.Payment.Amount);
            Logger.AppendKey("PaymentMethod", result.Payment.PaymentMethod);
            Logger.LogInformation("Payment created for rideId: {rideId} paymentId: {paymentId}",
                result.Payment.RideId, result.Payment.PaymentId);
        }
        
        return result;
    }
}

File: main workshop/services/python/payment-processor/payment_processor.py
"""
This file is part of the Payment Processor service.
Update this file in the Module 2 Idempotency exercise.
"""

from aws_lambda_powertools import Logger

from models import PaymentResult
from operations import PaymentOperations

logger: Logger = None


class PaymentProcessor:
    def __init__(self, logger_instance: Logger):
        global logger
        logger = logger_instance
        self.payment_ops = PaymentOperations()

    def handle_payment(self, event_detail: dict) -> PaymentResult:
        """Handle payment processing"""
        result = self.payment_ops.process_payment(event_detail)

        if result.success and result.payment:

            logger.info(
                "Payment created",
                extra={
                    "ride_id": result.payment.ride_id,
                    "payment_id": result.payment.payment_id,
                    "payment_amount": str(result.payment.amount),
                    "payment_method": result.payment.payment_method,
                },
            )

        return result
